src/components/Note/Note.component.tsx

```diff
@@ -1,12 +1,153 @@
-import React from 'react';
+import React, { useState, useEffect, useRef } from 'react';
+import { BiArchiveIn, BiArchiveOut, BiEdit, BiSave } from 'react-icons/bi';
+import { RiDeleteBinLine } from 'react-icons/ri';
+import { VscSymbolColor } from 'react-icons/vsc';
+import { BlockPicker } from 'react-color';
 
-import { NoteWrapper } from './Note.styled';
+import { useNotesContext } from '../../providers/Notes';
+import useOnClickOutside from '../../hooks/useOnClickOutside';
+import { NoteProps } from '../../utils/types';
+import { NoteWrapper, NoteButton, SaveButton, ErrorAlert } from './Note.styled';
+
+function Note({ id, title, description, bgColor }: NoteProps) {
+  const {
+    updateNote,
+    deleteNote,
+    addArchiveNote,
+    removeArchiveNote,
+    isArchived,
+    updateArchivedNote,
+  } = useNotesContext();
+  const noteRef = useRef<HTMLDivElement | null>(null);
+  const titleRef = useRef<HTMLInputElement>(null!);
+  const [noteTitle, setNoteTitle] = useState(title);
+  const [noteDescription, setNoteDescription] = useState(description);
+  const [selectedColor, setSelectedColor] = useState(bgColor);
+  const [openColorPicker, setOpenColorPicker] = useState(false);
+  const [editMode, setEditMode] = useState(false);
+  const [error, setError] = useState('');
+
+  const selectedNote = { id, title, description, bgColor };
+
+  const handleUpdateNote = () => {
+    setError('');
+    const title = noteTitle;
+    const description = noteDescription;
+    const bgColor = selectedColor;
+
+    if (title.trim().length > 0 && description.trim().length > 0) {
```

Comment:
What if we add the opposite condition to do an early return. Check out this [example](https://forum.freecodecamp.org/t/the-return-early-pattern-explained-with-javascript-examples/19364).

Reply:
Thank you for the tip!